nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -322,77 +322,68 @@ def add_electrodes(
+            else:
+                if prop == "brain_area":
+                    prop = "location"
+                elif prop == "group":
+                    if "group_name" not in property_names:
+                        prop = "group_name"
+                    else:
+                        continue
+                index = isinstance(data[0], (list, np.ndarray))
+                elec_columns[prop].update(description=prop, data=data, index=index)
 
+    # Fill with provided custom descriptions
     for x in metadata["Ecephys"]["Electrodes"]:
-        elec_columns[x["name"]]["description"] = x["description"]
         if x["name"] not in list(elec_columns):
-            raise ValueError(f'"{x["name"]}" not a property of se object')
+            raise ValueError(f'"{x["name"]}" not a property of se object, set it first and rerun')
+        elec_columns[x["name"]]["description"] = x["description"]
 
-    # updating default arguments if electrodes table already present:
+    # Updating default arguments if electrodes table already present:
     default_updated = dict()
     if nwbfile.electrodes is not None:
         for colname in nwbfile.electrodes.colnames:
             if colname != "group":
                 samp_data = nwbfile.electrodes[colname].data[0]
-                default_datatype = [
-                    proptype for proptype in channel_property_defaults if isinstance(samp_data, proptype)
-                ][0]
-                default_updated.update({colname: channel_property_defaults[default_datatype]})
+                default_datatype = [proptype for proptype in property_default_types if isinstance(samp_data, proptype)][
+                    0
+                ]
+                default_updated.update({colname: property_default_types[default_datatype]})
     default_updated.update(defaults)
 
+    # Re-shape data to have as many entries as channels are available in the table
     for name, des_dict in elec_columns.items():
         des_args = dict(des_dict)
         if name not in default_updated:
             if nwbfile.electrodes is None:
                 nwbfile.add_electrode_column(name=name, description=des_args["description"], index=des_args["index"])
             else:
-                combine_data = [channel_property_defaults[found_property_types[name]]] * len(nwbfile.electrodes.id)
-                des_args["data"] = combine_data + des_args["data"]
+                samp_data = des_dict["data"][0]
+                data_type_found = [proptype for proptype in property_default_types if isinstance(samp_data, proptype)][
+                    0
+                ]
+                extended_data = [property_default_types[data_type_found]] * len(nwbfile.electrodes.id)
+                previous_data = des_args["data"].tolist()
+                # Cast data to float
+                if data_type_found == Real:
+                    previous_data = [float(x) for x in previous_data]
```

Comment:
```suggestion
                if data_type_found == Real:
                    previous_data = [float(x) for x in previous_data]
```
Most of the comment insertions above are somewhat useful to enhancing readability, but this is self-apparent.

Reply:
Agree on this one.